### ISSUE TITLE
Add optimisation to with-own-theme function

### ISSUE THREAD

**Issue by mutebg** [newscorp-ghfb/NewsKit#751]
# Description
In the spike investigation about theme switching performance, we found that `with-own-theme` function takes too much time, in order to improve that we can implement a cache like in the proposed PR https://github.com/newscorp-ghfb/NewsKit/pull/706

This cache should be optional, we can add it as a `feature flag` the same way we do for CSS variables, in that way we can test on our web-site for the next-couple of months if that works correctly.

# Tasks/Steps
- copy the improved version of with-own-theme from https://github.com/newscorp-ghfb/NewsKit/pull/706
- add a feature flag `useThemeCache` ( or come up with better name ) and add the flag to theme-provider, and as a condition inside with-own-theme
- enable the flag to newskit website 

# Acceptance Criteria
*All acceptance criteria must be met before task can be marked as 'done'*

# Supporting Information
Will release without any supporting documentation for now

# Dependencies
*Any known business dependencies or reasons to engage with another team and input into their roadmap/timings*

**Comment by JohnTParsons** [comment]
URL for before testing: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/730-video-player-storybook/
(before fix merged, used http://ncu-newskit-docs-dev.s3-website-eu-west-1.amazonaws.com/components/drawer/)
URL for after testing: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/751-with-own-theme/components/drawer/

The first round of testing was done using the dev tools performance panel (see devtoolsperformance_before.png). Just doing a short recording against the docs site Drawer page, ie start record, open drawer, end record. In that way is easy to spot the things. What stood out, in this case, was “interaction type: click” which took 2.25s before the update.

![Image](https://user-images.githubusercontent.com/16664749/233087724-d0b2f0ad-0c1e-456d-b161-264a0d562b94.png)

The second round of testing, used React Profiler on local dev env (yarn dev:docs), as that is what Dow Jones use. The screenshots (reactprofiler_before.png and reactprofiler_after.png) show that rendering is 3x faster. It went from 1.239s to 0.4191s.


![Image](https://user-images.githubusercontent.com/16664749/233087927-63cd7ed2-98a9-495c-bb23-598a56e1082b.png)


![Image](https://user-images.githubusercontent.com/16664749/233087979-ea7bddcc-a289-4e9a-914e-f5e6520e4827.png)

**Comment by JohnTParsons** [comment]
Fixed in alpha version newskit@0.0.0-f8544123a (instead of newskit@7.2.0), which contains the fix. To turn it on, set useThemeCache to true inside NewsKitProvider like so:

```
<NewsKitProvider
     theme={theme}
     themeOptions={{exposeCssVariables: true, useThemeCache: true}}
```

In a future release, this might be turned on by default. (edited)
